### DISCUSSION TITLE
Run shell script when rip is finished

### DISCUSSION THREAD

**Discussion by thompcha** [automatic-ripping-machine/automatic-ripping-machine#628] [Q&A]
Greetings, thanks for this incredible project! I'm really excited about it!

I'm making use of Apprise and Pushbullet, but I would really like to know how I can run a shell script when a rip job is completed. Is this possible?

**Comment by ksimek** [comment]
Looks like the "BASH_SCRIPT" config parameter can be set to a path to a bash script, and the notification title will be received in $1 and body will be in $2

reference: 
https://github.com/automatic-ripping-machine/automatic-ripping-machine/blob/3f4cd7cbdc3ad1d9c9a7bb9d47d913d53e1688cd/arm/ripper/utils.py#L84

Excerpt:
```
def bash_notify(cfg, title, body):
    # bash notifications use subprocess instead of apprise.
    if cfg['BASH_SCRIPT'] != "":
        try:
            subprocess.run(["/usr/bin/bash", cfg['BASH_SCRIPT'], title, body])
            logging.debug("Sent bash notification successful")
        except Exception as error:  # noqa: E722
            logging.error(f"Failed sending notification via bash. Continuing  processing...{error}")
```

**Reply by YoSiJo** [reply]
It would be good if information from the job is passed to the script via ENV variables:

ARM_DEVICE=/dev/sr0
ARM_PATH=/home/arm/music/foo
ARM_RIPPER=localhost
ARM_RIPMETHOD=mkv
ARM_START=...
ARM_TIME=...
ARM_JOBTIME=...
...

The path would of course be very helpful.
